transport/transport-runtime/src/main/java/com/google/android/datatransport/runtime/backends/BackendFactory.java

```diff
@@ -0,0 +1,20 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.android.datatransport.runtime.backends;
+
+/** Used by backend discovery to create {@link TransportBackend}s. */
+public interface BackendFactory {
```

Comment:
We don't have components here, this library does not depend on firebase in any way by design.

Not sure what you mean by interop mechanism, what hard dependency are you trying to avoid?